src/ui/PackageDependencyProvider.ts

```diff
@@ -257,7 +257,7 @@ export class PackageDependenciesProvider implements vscode.TreeDataProvider<Tree
         const workspacePath = folderContext.folder.fsPath;
 
         const showingDependencies: Set<string> = new Set<string>();
-        const stack: Dependency[] = rootDependencies;
+        const stack: Dependency[] = rootDependencies.slice();
```

Comment:
Why is a shallow copy needed here?

Reply:
So this was a legit bug in the source, since popping below the dependency was lost, so subsequent times fetching children resulted in missing dependencies. Basically refreshes to the tree can cause stuff to be missed